### DISCUSSION TITLE
Unable to build the solution

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#16758] [Q&A] [answered]
_From the [addin Q&A](https://marketplace.visualstudio.com/items?itemName=unoplatform.uno-platform-addin-2022&ssr=false#qna)_

Hi, I tried to create the project via command line but I get errors, it is as if it did not find the necessary dependencies of Uno. I have checked the extensions and Uno Platform is found. What I could find is that the problem seems to be with the implementation with .Net 8, because I did a test with Uno Platform App (.Net 6, UWP) and I created it, I updated some packages, and the Xamarin SDK, I also enabled the Developer Mode on my PC and I could compile without problems.

**Comment by jeromelaban** [comment]
Could you post the errors you're getting from the command line? Thanks!

**Comment by SpiderArrow** [comment]
This is the command I'm running:
dotnet new unoapp -o unoapp2024 -preset "recommended" -markup "csharp" -platforms "maccatalyst" -platforms "windows" -platforms "desktop" -theme-service  -pwa False -theme "material" -presentation "mvux" -config False -di False -log "none" -nav "blank" -http False -loc False -tests "none" -toolkit  -dsp  -id "com.companyname.unoapp2024" -pub "O=unoapp2024"
and these are the errors: 

dotnet : Error: Opci├│n u opciones no v├ílidas:
En línea: 1 Carácter: 1
+ dotnet new unoapp -o unoapp2024 -preset "recommended" -markup "csharp ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Error: Opci├│n u opciones no v├ílidas::String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
-preset
   "-preset" no es una opci├│n v├ílida
recommended
   "recommended" no es una opci├│n v├ílida
-markup
   "-markup" no es una opci├│n v├ílida
csharp
   "csharp" no es una opci├│n v├ílida

**Reply by jeromelaban** [reply]
Thanks for the update. Have you run [uno-check](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html)?

**Reply by agneszitte** [reply]
I am not sure if you have already checked all the details for our get-started as well, but sharing it just in case:
https://aka.platform.uno/get-started

Are you able to create a project properly with the blank preset? (to check if there is a difference Vs the recommended preset).
![image](https://github.com/unoplatform/uno/assets/16295702/3e37841c-d40d-4cda-a2c1-6d15f5cbf7a3)

Have you tried also with these commands?
`dotnet new unoapp -preset=blank -o AppName`
`dotnet new unoapp -preset=recommended -o AppName`
(cf. https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows#uno-platform-application)

**Comment by SpiderArrow** [comment]
I followed the steps to use uno-check and thanks to it I found installation problems, I followed the steps on the page but still it does not create the solution correctly. I tried to repair my VS 2022 but it did not work. Try to use VS Code and run the command : 

```
dotnet new unoapp -preset=blank -o NEWAPP
```

But I got this information on the OUTPUT: 
```
2024-05-17 17:51:25.744 [error] (C:\Users\USUARIO\Desktop\newApp\NEWAPP\NEWAPP\NEWAPP.csproj): Could not resolve SDK "Uno.Sdk". Exactly one of the probing messages below indicates why we could not resolve the SDK. Investigate and resolve that message to correctly specify the SDK.
  SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" returned null.
  Unable to resolve 'Uno.Sdk (= 5.2.132)' for '.NETStandard,Version=v0.0'. PackageSourceMapping is enabled, the following source(s) were not considered: Microsoft Visual Studio Offline Packages, NuGet official package source.
2024-05-17 17:51:25.775 [error] Failed to load project 'C:\Users\USUARIO\Desktop\newApp\NEWAPP\NEWAPP\NEWAPP.csproj'. One or more errors occurred. (The SDK 'Uno.Sdk' specified could not be found.  C:\Users\USUARIO\Desktop\newApp\NEWAPP\NEWAPP\NEWAPP.csproj)
```

The extensions are installed, so I don't understand the error.
![image](https://github.com/unoplatform/uno/assets/38484006/34461227-a4d2-4f5e-99b9-b07de39d4185)

**Reply by jeromelaban** [reply]
Thank you for the update. 

Once the library has been created, even if there are errors, could you [create a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)? In your case, it's mostly doing `dotnet build /bl` in the folder, then posting the binlog file here.

**Reply by ToddGlodek** [reply]
Have y'all ever considered changing the [New Project configuration utility](https://marketplace.visualstudio.com/items?itemName=unoplatform.uno-platform-addin-2022) so that it is cloud based & runs inside a browser using WASM ??  It seems people constantly have difficulty creating projects - especially since it requires having you run [uno-check](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html) to keep the Templates up to date.   It seems like running the new project setup wizard in the cloud ( rather than as a Visual Studio extension, for example) would streamline the entire process by ensuring that all the latest templates/netgets/etc were guaranteed every time I created a new project.

I, for one, would prefer downloading a ZIP achieve containing all the per project files.

**Reply by jeromelaban** [reply]
Thanks for the suggestion @ToddGlodek. This would not change much, as the creation of the project is running fine, it's building that causes issues here. This specific error happens after the project is created, during the restore.

We still need to know why this error happens, even if it's a pure .NET SDK issue.

**Reply by ToddGlodek** [reply]
Obviously, you know best - I just thought it might be a super cool way to demonstrate the WASM capabilities of Uno.  Ya-know...  going right to the Uno Platform Home/Getting Started page...  running the project wizard in the browser... getting a  Zip achieve download in a few seconds without having to go through the [uno-check](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html) setup process.  

For a new user investigating Uno Platform for the first time, it would eliminate a lot of hurdles.  They could immediately see a familiar looking project structure that works on many different platforms - without having to do ANY previous development machine configuration/installation/customization.  You could also include the "Next Steps: Configuring Your Machine instructions" inside the Zip file as well.

**Reply by francoistanguay** [reply]
ICYMI, the first half of this is done here: [https://new.platform.uno](https://new.platform.uno/)

**Reply by ToddGlodek** [reply]
Yes, I have tried that before...   But

1. There does not seem to be any obvious way to get to [new.platform.uno](https://new.platform.uno/)  from the [Uno Home Page](https://platform.uno/).  The only place you can go is the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=unoplatform.uno-platform-addin-2022)
2. A new user would still have to go through the [uno-check](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html) setup process before pasting dotnet new command into the terminal.  If they missed/ignored the "Our Documentation" instructions, and didn't do that first, it would simply error out.  Whereas, if they got back a Zip Archive - they would have some immediate success and would be encouraged to continue exploring Uno Platform.
3. ![UnoPlatformNew](https://github.com/unoplatform/uno/assets/19764509/742d37c5-c402-4d0c-b3e0-8e3d62ad5db7)

**Reply by ToddGlodek** [reply]
@jeromelaban - Here's another argument in favor of downloading a ZIP achieve containing all the project files ( rather than copy-pasting a 'dotnet new' command).

Please see the following [post related uninstalling old template versions](https://github.com/unoplatform/uno/discussions/17355):

> I cannot create a new blank Uno project because of the conflicts with the older (v 4.7.56) versions of the templates. I tried to delete/re-install the latest versions of the templates - but cannot install the new templates because "templates use the same identity" as the old templates.

**Comment by SpiderArrow** [comment]
Hi, I followed the steps to create a binlog. Honestly it is the first time that I see files of this type, from what I can understand there are some tools that are being used that are being omitted, for practical purposes I attach the zip file with the solution that I generated through VS 2022 including the binlog file. Something that I could notice digging in the files that are in the folder Solution Items, is that in the global.json file the value of "Uno.Sdk" is in the version 5.2.132 the extension is in the version 5.2.49.
[UnoApp4.zip](https://github.com/unoplatform/uno/files/15397159/UnoApp4.zip)

**Reply by jeromelaban** [reply]
Thanks for the update.

It is very curious that you're getting Uno.Sdk 5.2.49, as this is a version we never published on nuget.org. You find how to update the uno.sdk [here](https://platform.uno/docs/articles/upgrading-nuget-packages.html#projects-using-the-unosdk).

This would indicate that somehow, nuget installed either a dev build, or something else incorrect.

If you ran uno.check, you should have had the latest templates installed automatically, but just in case, could you run `dotnet new update ` on your command line? Can you also paste the output of that command, as it will tell which version of the templates were previously installed. It may help determine where the package you got comes from.

**Reply by jeromelaban** [reply]
@SpiderArrow Thanks (please make sure to answer in the same thread, it will be easier for other community members to follow through).

This particular message (`PackageSourceMapping está habilitado`) seems to indicate that there's a `nuget.config` file that prevents the download of the `Uno.Sdk` package.

Let's try a few things:

1. Could you paste the contents of the file `%appdata%\NuGet\NuGet.Config` here? I suspect that there's a configuration that prevents nuget restore from working properly. Make sure to review the contents before posting it here, in case there's private information.
1. Can you try adding the following in a file named `nuget.config` next to the sln of the project that was created:
    ```xml
    <?xml version="1.0" encoding="utf-8"?>
    <configuration>
      <packageSources>
        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
      </packageSources>
    </configuration>
     ```
     Then try building the app with `dotnet build /bl` ?

**Reply by SpiderArrow** [reply]
Hello again, thank you for your reply. Here I share the Nuget.config file that I have on my computer. 
[NuGet.txt](https://github.com/unoplatform/uno/files/15419382/NuGet.txt)
I added the code in my nuget.config file in my project sln and then run the `dotnet build /bl` command and got this result: 
```
C:\Program Files\dotnet\sdk\8.0.300\NuGet.targets(169,5): warning : Unable to find a project to restore! [C:\Users\USUARIO\source\repos\UnoApp4\Uno
App4\UnoApp4.sln]

Compilación correcta.

C:\Program Files\dotnet\sdk\8.0.300\NuGet.targets(169,5): warning : Unable to find a project to restore! [C:\Users\USUARIO\source\repos\UnoApp4\Uno
App4\UnoApp4.sln]
    1 Advertencia(s)
    0 Errores

Tiempo transcurrido 00:00:00.50
```

**Reply by jeromelaban** [reply, marked as answer]
Thanks. The global `nuget.config` that you have for your system is very complex, and it feels like there's something wrong with its contents.

The default one when installing a new machine should look like this:
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="Microsoft Visual Studio Offline Packages" value="C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\" />
  </packageSources>
  <packageRestore>
    <add key="enabled" value="True" />
    <add key="automatic" value="True" />
  </packageRestore>
  <bindingRedirects>
    <add key="skip" value="False" />
  </bindingRedirects>
  <packageManagement>
    <add key="format" value="0" />
    <add key="disabled" value="False" />
  </packageManagement>
</configuration>
```

In Visual Studio, without opening a solution, could you:
- Go to `Tools`, `Options`
- Go to `NuGet Package Manager`
- Then in `Package Source Mapping`

Could you show a screenshot of all that you're seeing?
Then could you try clicking the `Remove All` button, click OK to close the option dialog, and try creating an uno project again?

Thanks for the help in troubleshooting this!

**Reply by SpiderArrow** [reply]
EUREKA! 
That was the problem, I was able to create the project without errors!
Thank you very much for the support, I will continue to explore the platform.
![Hello-Uno-Platform](https://github.com/unoplatform/uno/assets/38484006/3ef81236-3e63-429d-bc94-f8bc7da89bca)

**Reply by jeromelaban** [reply]
Thank you! This is very valuable. We'll certainly extend uno.check to validate for those scenarios.

**Reply by agneszitte** [reply]
(Related issue for Uno.Check: https://github.com/unoplatform/uno.check/issues/256)

**Comment by SpiderArrow** [comment]
I followed the steps to update Uno but didn't have much luck fixing the reference to the Uno Sdk. Install dotnet-outdated, run the command in the folder where the .sln file is and in the subfolder where the .csproj file is. I also ran the dotnet new update command but it tells me that the packages are updated. I attach the results of the console in the txt file. 
[test-dotnet-outdated.txt](https://github.com/unoplatform/uno/files/15410290/test-dotnet-outdated.txt)
